net/netmon/netmon_darwin.go

```diff
@@ -56,7 +56,19 @@ func (m *darwinRouteMon) Receive() (message, error) {
 		if err != nil {
 			return nil, err
 		}
-		msgs, err := route.ParseRIB(route.RIBTypeRoute, m.buf[:n])
+		msgs, err := func() (msgs []route.Message, err error) {
+			defer func() {
+				// #14201: permanent panic protection, as we have been burned by
+				// ParseRIB panics too many times.
+				msg := recover()
+				if msg != nil {
+					msgs = nil
+					m.logf("[unexpected] netmon: panic in route.ParseRIB from % 02x", m.buf[:n])
+					err = fmt.Errorf("panic in route.ParseRIB: %s", msg)
+				}
+			}()
+			return route.ParseRIB(route.RIBTypeRoute, m.buf[:n])
+		}()
 		if err != nil {
 			if debugRouteMessages {
```

Comment:
shouldn't this tear down the `m.fd` and re-open it? It seems like once this returns an error, our framing might be off forever?

Reply:
perhaps yes, it's not clear if the issue afoot is a framing issue, but this is a good idea.

to do so requires rewriting the concurrency mechanics around the fd/close, so i'll have to look at it in a separate change.

Reply:
https://github.com/tailscale/tailscale/pull/14888